snuba/clusters/cluster.py

```diff
@@ -151,6 +151,7 @@ def get_node_connection(
         user: str,
         password: str,
         database: str,
+        fallback_host: Optional[str] = None,
```

Comment:
Who is providing the fallback host ?
We cannot rely on it being hardcoded in settings otherwise we would have to change production settings to add/remove hosts, which would defy the purpose of the PR.

I think you should have two runtime config flags: 
- one to activate/deactivate the fallback
- another (string type) with a comma separated list of hosts.

When you instantiate a cluster (which happens at startup), pick one random host from the runtime setting. 

Alternative approach: Avoid making change to cluster.py entirely. and do everything in native.py:
- When you need a fallback host pick a random one from the runtime settings, create the connecitons and store it in the queue.
- If a new query shows up that needs the fallback host, but the existing connection is already in use, pick another host from the runtime setting and add it to the queue. 

THis is just like the current connection pool works. it keeps a queue of connections (thread safe) every time a query is executed, it tries to pop a conneciton from the queue, if there is none, it creates one, once the query is done it puts the conneciton back in the queue.

Reply:
In constructing the runtime config key for the list of fallback hosts, do you think it makes sense to use:

`fallback_hosts:<database_name>`

because then I see a clear path to getting the list from within ClickhousePool without needing to pass more information from cluster.py on creation.